c++/src/capnp/schema-loader.c++

```diff
@@ -1263,7 +1331,10 @@ _::RawSchema* SchemaLoader::Impl::load(const schema::Node::Reader& reader, bool
     // Prefer to replace the existing schema if the existing schema is a placeholder.  Otherwise,
     // prefer to keep the existing schema.
     shouldReplace = checker.shouldReplace(
-        existing, validatedReader, schema->lazyInitializer != nullptr);
+        existing, validatedReader, schema->lazyInitializer != nullptr, replacementExpectedToBeNewer);
```

Comment:
Hmm, it seems to me that there should be no need to pass a new flag down to `CompatibilityChecker`. The class already decides which schema is "newer", so all we need to do is then verify that the replacement was deemed newer or equivalent.

Why do we need any deeper changes to the checker? Is it just to improve the error message in the case of removed fields?